src/performance/gpu/amd/asus.rs

```diff
@@ -38,58 +34,131 @@ impl ASUS {
+
+                match platform.set_ppt_fppt(self.tdp + self.boost).await {
+                    Ok(()) => Ok(()),
+                    Err(_) => {
+                        self.set_tdp_boost_wmi().await
                     },
+                }
+            },
+            Err(_) => {
+                log::warn!("Unable to use asusd to read tdp, asus-wmi interface will be used");
+                self.set_tdp_boost_wmi().await
+            }
+        }
+    }
+
+    async fn set_tdp_boost_wmi(&self) -> TDPResult<()> {
+        match self.platform.lock() {
+            Ok(platform) => {
+                match platform.set_ppt_fppt(self.tdp + self.boost) {
+                    Ok(()) => Ok(()),
                     Err(err) => {
-                        log::warn!("Error fetching ppt_apu_sppt: {}", err);
+                        log::warn!("Unable to set ppt_fppt: {}", err);
                         Err(TDPError::FailedOperation(format!("")))
                     }
                 }
             },
-            Err(err) => {
+            Err(_) => {
+                log::warn!("Unable to use asus-wmi interface");
+                Err(TDPError::FailedOperation(format!("")))
+            },
+        }
+    }
+
+    async fn set_throttle_thermal_policy(&self, throttle_policy: ThrottlePolicy) -> TDPResult<()> {
+        match RogDbusClient::new().await {
+            Ok((dbus, _)) => {
+                let platform = dbus.proxies().rog_bios();
+
+                match platform.set_throttle_thermal_policy(throttle_policy).await {
+                    Ok(()) => Ok(()),
+                    Err(_) => {
+                        log::warn!("Unable to use asusd to read tdp, asus-wmi interface will be used");
+                        Err(TDPError::FailedOperation(format!("")))
+                    },
+                }
+            },
+            Err(_) => {
                 log::warn!("Unable to use asusd to read tdp, asus-wmi interface will be used");
                 Err(TDPError::FailedOperation(format!("")))
             }
         }
     }
+}
+
+impl TDPDevice for ASUS {
+    async fn tdp(&self) -> TDPResult<f64> {
+        Ok(self.tdp.into())
+    }
 
     async fn set_tdp(&mut self, value: f64) -> TDPResult<()> {
-        todo!()
+        self.tdp = value.round() as u8;
+
+        self.set_tdp_boost().await
     }
 
     async fn boost(&self) -> TDPResult<f64> {
-        todo!()
+        Ok(self.boost.into())
     }
 
     async fn set_boost(&mut self, value: f64) -> TDPResult<()> {
-        todo!()
+        self.boost = value.round() as u8;
+
+        self.set_tdp_boost().await
     }
 
     async fn thermal_throttle_limit_c(&self) -> TDPResult<f64> {
-        todo!()
+        Ok(0.0)
     }
 
-    async fn set_thermal_throttle_limit_c(&mut self, limit: f64) -> TDPResult<()> {
-        todo!()
+    async fn set_thermal_throttle_limit_c(&mut self, _limit: f64) -> TDPResult<()> {
+        Ok(())
     }
```

Comment:
I need some input on what to do about these, I couldn't find anything in asusctl that lets me limit the max temp

Reply:
Perhaps we could fallback to the ryzenadj methods for this?

Reply:
Maybe.. It would be bad if ryzenadj starts overriding other vars though, I'll see if i can call the ryzenadj function from there

Reply:
I'm now using TDP struct in the `ASUS` struct as an `Arc<Mutex<TDP>>` and calling `self.ryzenadj.lock().await` to interact with RyzenAdj